### PR TITLE
Fix the IPA extractor issue from the newly added IPA file.

### PR DESCRIPTION
**Pull request by buptkang**
During the pressure testing, the packager failed to extract a IPA file, which has been included in this PR. This PR fixed this issue.

This PR do not resolve the unit testing issue that has been reviewing in the other PR.

### CODE REVIEW

File: src/contentBase.ts [hunk 1]
```diff
@@ -77,10 +77,14 @@ export class ContentBase {
...
-                            await fse.open(buildPath, 'w+');
-                            var wstream = fse.createWriteStream(buildPath);
-                            readStream.pipe(wstream);
+                            try {
+                                await fse.ensureFile(buildPath);
+                                await fse.open(buildPath, 'w+');
+                                var wstream = fse.createWriteStream(buildPath);
+                                readStream.pipe(wstream);
+                            } catch(error) {
+                                zipfile.readEntry();
```
Comment (izikl) [inline src/contentBase.ts]: Some background: the reason for this change is a test failure with a certain IPA file. 
Root cause: yauzl returned a folder that matched the searchStrings. So saving directory as a file cause the failure. 
Suggested solution: ignore any failures when saving folder AND FILES.

Client that use this library should know if the app-metadata failed to extract the metadata before zip failed when saving one of the files, so I would prefer to avoid this solution. 

Instead I would just suggest to skip folders in the first place. 
Basically changing this line:
https://github.com/Microsoft/app-metadata/blob/master/src/contentBase.ts#L70
to this line:
``` if (valuable && !validName.endsWith(path.sep)) { ```

**Comment by buptkang** [comment]
👍
